etcdserver/server.go

```diff
@@ -351,6 +354,10 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 		cl.SetStore(st)
 		cl.SetBackend(be)
 		cl.Recover()
+		if cl.Version != nil && cl.Version().LessThan(semver.Version{Major: 3}) && !beExist {
+			os.RemoveAll(bepath)
```

Comment:
this seems racey-- delete the file, bring up etcd, let the backend create a new bepath, then kill etcd before it reaches this point, start etcd again and it'll boot